nova_vm/src/engine/small_double.rs

```diff
@@ -0,0 +1,134 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+/// 56-bit double, the implied bottom 8 bits are zero.
+#[derive(Clone, Copy, PartialEq)]
+pub struct SmallF64 {
+    data: [u8; 7],
+}
+
+impl std::fmt::Debug for SmallF64 {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "{}", Into::<f64>::into(*self))
+    }
+}
+
+impl SmallF64 {
+    pub(crate) const fn _def() -> SmallF64 {
+        Self {
+            data: [1, 2, 3, 4, 5, 6, 7],
+        }
+    }
+
+    #[inline]
+    pub fn into_f64(self) -> f64 {
+        self.into()
+    }
+
+    #[inline(always)]
+    fn can_convert(value: f64) -> bool {
+        value.to_bits().trailing_zeros() >= 8
```

Comment:
Should this also check that `value` is not a valid `SmallInteger`?

Reply:
Yeah, sounds like a good idea.